api/envoy/extensions/filters/listener/original_dst/v3/original_dst.proto

```diff
@@ -17,4 +17,16 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message OriginalDst {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.listener.original_dst.v2.OriginalDst";
+  enum Method {
+    // Use sockopt SO_ORIGINAL_DST or IP6T_SO_ORIGINAL_DST to recover the original local address. This is the default method.
+    Dnat = 0;
```

Comment:
`DNAT`, etc.

Reply:
I'm actually confused about this feature as a user. How and when do I set these options? It looks like there is more than just recovery, e.g. "mark the local address as restored"..

Reply:
> It looks like there is more than just recovery, e.g. "mark the local address as restored"
Exactly! DNAT will do both "recover from sockopt" and "mark as restored". That's the current filter's behavior. 
I just name it DNAT...

Inspired by the @tomastigera 's experimental commit in #16130 
In TPROXY-and-no-DNAT iptable rules, the original_dst_cluster need the latter "mark as restored" flag. That's what the second method does.
Some integration tests will also find this behavior useful.

For anybody else, either explicitly set DNAT or use the default value DNAT.

Reply:
OK, but "as a reader" I have no idea how to use this still. This is user facing documentation and needs to spell out what these features are and link to references and/or Envoy documentation as appropriate.

Reply:
Make sense to me. WIll update

Reply:
I think this does not need to dive into the SO_ORIGINAL_DST detail and rather stress that this is the method that needs to be used with REDIRECT redirection.